Laboratory/C++/test/enum_size.cpp

```diff
@@ -0,0 +1,15 @@
+#include "../gen/enum_size.hpp"
+#include <cstdio>
+
+int main() {
+  if (sizeof(SmallEnum) != sizeof(uint8_t)) {
```

Comment:
Same comment here, this should be checking fixed size `1` and `8` and I do worry slightly more with C++.

Reply:
The thing is that on a system where `sizeof(uint8_t) == 4` or something, I am pretty sure that defining `enum SmallEnum : uint8_t` (as our codegen does) will also make `sizeof(SmallEnum) == 4`.